mlflow/pipelines/steps/train.py

```diff
@@ -887,7 +946,7 @@ def objective(X_train, y_train, validation_df, hyperparameter_args, on_worker=Fa
     def _log_estimator_to_mlflow(self, estimator, X_train_sampled, on_worker=False):
         from mlflow.models.signature import infer_signature
 
-        if hasattr(estimator, "best_score_"):
+        if hasattr(estimator, "best_score_") and (type(estimator.best_score_) in [int, float]):
```

Comment:
What other types could `best_score_` be? None?

Reply:
The issue is that in some cases (lgbm as far as I remember) it can be a Dict. It is not related to AutoML, I have seen this error before as well, with AutoML it's just easier to reproduce it.